src/application/validation/validators/email-validation.ts

```diff
@@ -0,0 +1,17 @@
+import { type EmailValidator } from '@/application/validation'
+import { type Validation } from '@/application/protocols'
+import { InvalidParamError } from '@/application/errors'
+
+export class EmailValidation implements Validation {
+  constructor (
+    private readonly fieldName: string,
+    private readonly emailValidator: EmailValidator
```

Comment:
It is a bit confuse an EmailValidaton class has an EmailValidator dependency, a slightly difference. Maybe a suffixe Impl/Adapter to clarify.

Reply:
how about just validator?

Reply:
That is good.